datafusion/physical-expr/src/array_expressions.rs

```diff
@@ -1821,88 +1822,95 @@ pub fn array_has_all(args: &[ArrayRef]) -> Result<ArrayRef> {
-        (DataType::Boolean, DataType::Boolean) => array_intersect_normal!(first_array, second_array, DataType::Boolean, BooleanArray, BooleanBuilder),
-        // (DataType::Float32, DataType::Float32) => array_intersect_normal!(arr, element, Float32Array),
-        // (DataType::Float64, DataType::Float64) => array_intersect_normal!(arr, element, Float64Array),
-        (DataType::Int8, DataType::Int8) => array_intersect_normal!(first_array, second_array, DataType::Int8, Int8Array, Int8Builder),
-        (DataType::Int16, DataType::Int16) => array_intersect_normal!(first_array, second_array, DataType::Int16, Int16Array, Int16Builder),
-        (DataType::Int32, DataType::Int32) => array_intersect_normal!(first_array, second_array, DataType::Int32, Int32Array, Int32Builder),
-        (DataType::Int64, DataType::Int64) => array_intersect_normal!(first_array, second_array, DataType::Int64, Int64Array, Int64Builder),
-        (DataType::UInt8, DataType::UInt8) => array_intersect_normal!(first_array, second_array, DataType::UInt8, UInt8Array, UInt8Builder),
-        (DataType::UInt16, DataType::UInt16) => array_intersect_normal!(first_array, second_array, DataType::UInt16, UInt16Array, UInt16Builder),
-        (DataType::UInt32, DataType::UInt32) => array_intersect_normal!(first_array, second_array, DataType::UInt32, UInt32Array, UInt32Builder),
-        (DataType::UInt64, DataType::UInt64) => array_intersect_normal!(first_array, second_array, DataType::UInt64, UInt64Array, UInt64Builder),
+        (DataType::Utf8, DataType::Utf8) => DataType::Utf8,
+        (DataType::LargeUtf8, DataType::LargeUtf8) => DataType::LargeUtf8,
+        (DataType::Boolean, DataType::Boolean) => DataType::Boolean,
+        (DataType::Float32, DataType::Float32) => DataType::Float32,
+        (DataType::Float64, DataType::Float64) => DataType::Float64,
+        (DataType::Int8, DataType::Int8) =>  DataType::Int8,
+        (DataType::Int16, DataType::Int16) => DataType::Int16,
+        (DataType::Int32, DataType::Int32) => DataType::Int32,
+        (DataType::Int64, DataType::Int64) => DataType::Int64,
+        (DataType::UInt8, DataType::UInt8) => DataType::UInt8,
+        (DataType::UInt16, DataType::UInt16) => DataType::UInt16,
+        (DataType::UInt32, DataType::UInt32) => DataType::UInt32,
+        (DataType::UInt64, DataType::UInt64) => DataType::UInt64,
         // (DataType::Null, _) => return Ok(array(&[ColumnarValue::Array(args[1].clone())])?.into_array(1)),
-        (first_value_dt, second_value_dt) => {
-            Err(DataFusionError::NotImplemented(format!(
+        (first_value_dt, second_value_dt) =>
+            return Err(DataFusionError::NotImplemented(format!(
                 "array_intersect is not implemented for '{first_value_dt:?}' and '{second_value_dt:?}'",
             )))
+    };
+
+    let mut offsets = vec![0];
+
+    let mut tmp_values = vec![];
+
+    let mut converter = RowConverter::new(vec![SortField::new(dt.clone())])?;
```

Comment:
:heart:

Reply:
haven't finished. ^ . ^ will be finished in hours..

Reply:
thanks for your advice !! very useful for me !